README.md

```diff
@@ -52,11 +52,11 @@ defmodule MyApp.Guardian do
   use Guardian, otp_app: :my_app
 
   def subject_for_token(resource, _claims) do
-    to_string(resource.id)
+    {:ok, to_string(resource.id)}
```

Comment:
I would feel a bit weird to be honest. Requiring an ok/error tuple provides a more robust API both internally and externally by being able to rely on the shape of the return. It also provides for the developer to determine what an error is/is not. By making the return type ambiguous we move the requirement to determine the ok/error status of the return to all callsites.

The example above is one of the more simple ones but I've used many that deal with straight up strings as well as calling out to an external service. I don't feel like requiring an ok/error tuple is a burden. I'm feeling like ambiguous returntypes would make dealing with this return type propagate to all callsites rather than keeping it focused.